src/windshaft/error.js

```diff
@@ -14,7 +14,7 @@ var WindshaftError = function (error) {
   }
   if (this.isAnalysisError(error.type)) {
     this.context = error.analysis && error.analysis.context;
-    this.analysisId = error.analysis && error.analysis.node_id;
+    this.analysisId = error.analysis && (error.analysis.node_id || error.analysis.id);
```

Comment:
Those identifiers come from the client. For instance in map instantiation:

```
"layers": [ ... ],
"dataviews": { ... },
"analyses": [
    {
        "id": "a1",
        "type": "buffer",
        "params": {
            "source": {
                "id": "a0",
                "type": "source",
                "params": {
                    "query": "SELECT * FROM populated_places_simple"
                }
            },
            "radius": 100000,
            "isolines": 1,
            "dissolved": false
        }
    }
]
```

If something went wrong in `source` node, camshaft tries to get the `id` from `buffer` (`a1`) node as `error.analysis.id` because it is the root node and the `id` from `source` (`a0`) as `error.analysis.node_id` because it is the failing node.

Camshaft does not enforce to have an `id` because it handles its own internal identifiers for each node and windshaft relies on `id`s that needs to replace the queries for layers in layergroup.

Said that, if you have a STR  when `error.analysis.node_id` is not populated, please, file a ticket in camshaft's repo.